WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift

```diff
@@ -70,11 +76,182 @@ struct NewOrder: View {
     }
 }
 
+// MARK: Order Sections
+/// Represents the Products section
+///
+private struct ProductsSection: View {
+    let geometry: GeometryProxy
+
+    var body: some View {
+        Group {
+            Divider()
+
+            VStack(alignment: .leading, spacing: NewOrder.Layout.verticalSpacing) {
+                Text(NewOrder.Localization.products)
+                    .headlineStyle()
+
+                // TODO: Add a product row for each product added to the order
+                ProductRow()
+
+                AddButton(title: NewOrder.Localization.addProduct) {
+                    // TODO: Open Add Product modal view
+                }
+            }
+            .padding(.horizontal, insets: geometry.safeAreaInsets)
+            .padding()
+            .background(Color(.listForeground))
+
+            Divider()
+        }
+    }
+
+    /// Represent a single product row in the Product section
+    ///
+    struct ProductRow: View {
+        var body: some View {
+            AdaptiveStack(horizontalAlignment: .leading) {
+                HStack(alignment: .top) {
+                    // Product image
+                    // TODO: Display actual product image when available
+                    Image(uiImage: .productPlaceholderImage)
+                        .aspectRatio(contentMode: .fill)
+                        .frame(width: NewOrder.Layout.productImageSize, height: NewOrder.Layout.productImageSize)
+                        .foregroundColor(Color(UIColor.listSmallIcon))
+                        .accessibilityHidden(true)
+
+                    // Product details
+                    VStack(alignment: .leading) {
+                        Text("Love Ficus") // Fake data - product name
+                        Text("7 in stock • $20.00") // Fake data - stock / price
+                            .subheadlineStyle()
+                        Text("SKU: 123456") // Fake data - SKU
+                            .subheadlineStyle()
+                    }
+                    .accessibilityElement(children: .combine)
+                }
+
+                Spacer()
+
+                ProductStepper()
+            }
+
+            Divider()
+        }
+    }
+
+    /// Represents a custom stepper.
+    /// Used to change the product quantity in the order.
+    ///
+    struct ProductStepper: View {
+        @State var textSize: CGSize = .zero
+
+        var body: some View {
+            HStack(spacing: textSize.height) {
+                Button {
+                    // TODO: Decrement the product quantity
+                } label: {
+                    Image(uiImage: .minusImage)
+                        .resizable()
+                        .aspectRatio(contentMode: .fit)
+                        .frame(height: textSize.height)
+                }
+                Text("1") // Fake data - quantity
+                    .background(ViewGeometry())
```

Comment:
Nice, I have never used `PreferenceKeys` before. I did something similar in [`scrollVerticallyIfNeeded`](https://github.com/woocommerce/woocommerce-ios/blob/3d48a920d7c26c89f9cf16a1e5b0a4fbe1d38db9/WooCommerce/Classes/View%20Modifiers/View%2BScrollModifiers.swift) but maybe this is the correct way! 

Do you think we should add some more documentation explaining why is this needed?

Reply:
Yeah, I learned there's a lot that can be done with `PreferenceKeys`! That said, conceptually what I really want to do here is mainly just scale everything as the text size changes. I wasn't aware that I could use `@ScaledMetric` to do that. The only real difference is how I set the initial size of everything (based on the text size or based on a constant).

I think rather than using `PreferenceKeys` in this case, and trying to document it to explain it further, I'll switch to using a scaled metric for sizing here.

Reply:
I tried it out and the `scaled metric` approach seems clearer to me; switched to that in bb33741.